gulpfile.js

```diff
@@ -0,0 +1,368 @@
+
+    let cmd = `echo '{ "uf_id": { "IN":[%{uids}] } }' | cv api UFMatch.get sequential=1`;
+    cmd = cmd.replace('%{uids}', _.map(map, 'drupal').join(','));
+
+    const ufMatches = JSON.parse(execSync(cmd, { cwd: civiDir })).values;
+
+    _.each(ufMatches, (ufMatch) => {
+      const user = _.find(map, user => user.drupal === ufMatch.uf_id);
+
+      user.civi = ufMatch.contact_id;
+    });
+  }
+}
+
+/**
+ * Runs backstopJS with the given command.
+ *
+ * It fills the template file with the list of scenarios, creates a temp
+ * file passed to backstopJS, then removes the temp file once the command is completed
+ *
+ * @param  {String} command
+ * @return {Promise}
+ */
+async function runBackstopJS (command) {
+  if (touchSiteConfigFile()) {
+    throwError(
+      'No site-config.json file detected!\n' +
+      `\tOne has been created for you in the root/\n` +
+      '\tPlease insert the real value for each placeholder and try again'
+    );
+  }
+
+  return new Promise((resolve, reject) => {
+    let success = false;
+
+    gulp.src(FILES.tpl)
+      .pipe(file(path.basename(FILES.temp), createTempConfig()))
+      .pipe(gulp.dest('.'))
+      .on('end', async () => {
+        try {
+          if (_.includes(['reference', 'test'], command) && shouldCreateCookies()) {
+            console.log('Writing cookies...');
+            await writeCookies();
+          }
+
+          await backstopjs(command, {
+            config: FILES.temp,
+            filter: argv.filter
+          });
+
+          success = true;
+        } finally {
+          cleanUpAndNotify(success);
+
+          success
+            ? resolve()
+            : reject(new Error('BackstopJS error. It may be a task error, a script error or a comparison error'));
+        }
+      });
+  })
+    .catch(err => {
+      throwError(err.message);
+    });
+}
+
+/**
+ * Gets the name of the selected group specified in the --group argument
+ * If the argument was not passed, uses the name of the default group instead
+ *
+ * @return {String}
+ */
+function selectedGroup () {
+  return argv.group ? argv.group : DEFAULT_GROUP;
+}
+
+/**
+ * Runs several criteria to check whether cookies should be (re)created
+ * before running the test suites
+ *
+ * @return {Boolean}
+ */
+function shouldCreateCookies () {
+  if (!fs.existsSync(DIRS.cookies)) {
+    return true;
+  }
+
+  const cookies = fs.readdirSync(DIRS.cookies);
+
+  if (!cookies.length) {
+    return true;
+  }
+
+  const everyUserHasCookies = USERS.every(user => {
+    return _.find(cookies, cookie => path.basename(cookie, '.json') === user);
+  });
+
+  if (!everyUserHasCookies) {
+    return true;
+  }
+
+  const userCookies = JSON.parse(fs.readFileSync(path.join(DIRS.cookies, cookies[0]), {
+    encoding: 'utf8'
+  }));
+  const sessionCookie = _.find(userCookies, cookie => _.startsWith(cookie.name, 'SESS'));
+  const isCookieExpired = sessionCookie.expires * 1000 <= Date.now();
```

Comment:
lets define `cookies, everyUserHasCookies, userCookies, sessionCookie,` and  `isCookieExpired`  at the top of the function.

Reply:
you can't, `const` doesn't work like `var`, you can't do this
```js
const userCookies;
const sessionCookie;
const isCookieExpired;

userCookies = // ...
sessionCookie = // ...
isCookieExpired = // ...
```